File: Komuku-Game/src/core/count/ScoreOne.scala
package core.count

import core.GameMap
import entity.Point
import enumeration.Color

object ScoreOne extends ScoreBase {
  override def getScore(point: Point): Int = {
    var value = 0
    if (GameMap.getColor(point) == Color.NULL) {
      return value
    }
    for (i <- 0 until 8) {
      val fresh = GameMap.getRelatePoint(point, i, 1)
      if (GameMap.reachable(fresh)) {
        if (GameMap.getColor(fresh) == ScoreData.currentColor)
          value += 2
        if (GameMap.getColor(fresh) == Color.getOtherColor(ScoreData.currentColor))
          value += 2
        if (GameMap.getColor(fresh) == Color.NULL)
          value += 1
      }
    }
    if (GameMap.getColor(point) != ScoreData.currentColor) {
      value = -value
    }
    value
  }
}

File: Komuku-Game/src/core/evalution/ScoreOne.java
package core.evalution;

import core.GameMap;
import entity.Point;
import enumeration.Color;

public class ScoreOne implements ScoreBase {

    @Override
    public int getScore(GameMap gameMap, Point point, Color color) {
        int value = 0;
        if (gameMap.getColor(point) == Color.NULL) {
            return value;
        }
        for (int i = 0; i < 8; i++) {
            Point fresh = gameMap.getRelatePoint(point, i, 1);
            if (gameMap.reachable(fresh)) {
                if (gameMap.getColor(fresh) == color)
                    value += 2;
                if (gameMap.getColor(fresh) == color.getOtherColor())
                    value += 2;
                if (gameMap.getColor(fresh) == Color.NULL)
                    value += 1;
            }
        }
        if (gameMap.getColor(point) != color) {
            value = -value;
        }
        return value;
    }

}
